src/app/sign-in/pages/sign-in/sign-in.component.ts

```diff
@@ -157,6 +192,33 @@ export class SignInComponent implements OnInit {
     this.badCredentials = false
   }
 
+  loadRequestInfoForm(): void {
+    this._oauthService.loadRequestInfoForm().subscribe((data) => {
+      if (data) {
+        this.requestInfoForm = data
+        if (this.requestInfoForm.userId) {
+          this.authorizationForm.patchValue({
+            username: this.requestInfoForm.userId,
+          })
+        } else {
+          if (
+            this.requestInfoForm.userEmail ||
+            this.requestInfoForm.userFamilyNames ||
+            this.requestInfoForm.userGivenNames
+          ) {
+            this._router.navigate(['/register'], {
+              queryParams: this.oauthParameters,
```

Comment:
If the redirect is required, this can also be done on the backend to avoid the extra step, so you might add a TODO here

Reply:
Yes, the old frontend redirects to the register, and not necessarily there can be a scenario when some request to someone else to create an account and give them access and they share the URL with the email and names.

> Does the old frontend also redirects all Oauth request that contains a `userEmail`, `userFamilyNames`, `userGivenNames` to the register right away?
> Because I think, that maybe even if the request comes with that data, the user might be already registered and just wants to sing in, but let's clone whatever the old frontend does.

Ok, I will add a TODO and I'll create an issue